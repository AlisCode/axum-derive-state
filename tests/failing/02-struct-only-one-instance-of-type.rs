//! Only one instance of a type is allowed in a struct
//! The error should provide a helpful hint by suggesting to create an appropriate newtype

use derive_state::State;

#[derive(State)]
pub struct MyState {
    database: String,
    http_client: String,
}

fn main() {}
