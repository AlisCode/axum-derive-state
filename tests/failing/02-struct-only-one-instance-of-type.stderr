error: Type String is used multiple times in the state

         = note: The state may only contain exactly one instance of a given type
         = help: Consider wrapping this type in a newtype, e.g. pub struct HttpClient(String);

 --> tests/failing/02-struct-only-one-instance-of-type.rs:9:5
  |
9 |     http_client: String,
  |     ^^^^^^^^^^^
